src/TerraformCommand.groovy

```diff
@@ -1,3 +1,26 @@
-interface TerraformCommand {
-    public String getEnvironment()
+trait TerraformCommand {
+    private String environment
+    private String terraformBinary = "terraform"
```

Comment:
Reaching into a trait and pulling out a private variable seems fraught with peril.  In fact, I think your change makes it clear that `terraformBinary` never belonged in TerraformCommand to begin with. It feels like this needs to be pushed out to a higher level class - I want to be able to change the terraform binary once, and have it cascade across all commands.

Reply:
We're not pulling out a private variable - we're going through the public `get/set` accessors. I would typically make ALL actual variables private and get to them via `get/set`.

Reply:
Whether or not you access the variable directly, or use a get/set has very little difference.  The point is that we're reaching into the internals of the class and operating on those values externally.  That fails to encapsulate *behavior*.  

There's no *behavior* in `TerraformCommand` that relies on `terraformBinary` - hence the need to reach in, and pull it out - because the behavior lives elsewhere (the behavior does not live in the trait itself).

Reply:
Removed terraformBinary from TerraformCommand